test/gtest/ucp/test_ucp_fence.cc

```diff
@@ -39,7 +40,8 @@ class test_ucp_fence : public ucp_test {
         void *request = ucp_atomic_fetch_nb(e->ep(), UCP_ATOMIC_FETCH_OP_FADD,
                                             *initial_buf, (T*)result_buf, sizeof(T),
                                             (uintptr_t)memheap_addr, rkey, send_cb);
-        request_wait(request);
+        std::vector<entity*> entities = {e};
+        request_wait(request, entities);
```

Comment:
It should not hang because the main thread executing `run_workers()` below is taking care of progressing the receiver side. 

Actually, I tried removing the `progress({&receiver()})` call in `run_workers()` and the tests properly hangs.